internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -252,6 +252,20 @@ func (r *ClusterDeploymentsReconciler) agentClusterInstallFinalizer(ctx context.
 		}
 	} else { // clusterInstall is being deleted
 		if funk.ContainsString(clusterInstall.GetFinalizers(), AgentClusterInstallFinalizerName) {
+			cluster, err := r.Installer.GetClusterByKubeKey(req.NamespacedName)
+			if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
```

Comment:
should handle also other errors? i.e. just if 'err != nil'

Reply:
If there is a error that is not ErrRecordNotFound we requeue.
If there is ErrRecordNotFound we ignore and continue the flow ( no need to check if cluster install need to be cancelled)